Tests/Creation/Frame/Content/String/ID3FrameFromBooleanContentCreatorTest.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  ID3FrameFromStringContentCreatorTest.swift
+//
+//  Created by Nolaine Crusher on 03/31/2020.
+//  2018 Fabrizio Duroni.
+//
+
+import XCTest
+@testable import ID3TagEditor
+
+class ID3BooleanFrameCreatorTest: XCTestCase {
+    func testCreatorFrameFromBooleanContent() {
+        let id3FrameFromBooleanContentCreator = ID3FrameFromBooleanContentCreator(
+                frameContentSizeCalculator: MockFrameContentSizeCalculator(),
+                frameFlagsCreator: MockFrameFlagsCreator()
+        )
+
+
+        let frameBytes = id3FrameFromBooleanContentCreator.createFrame(
+            frameIdentifier: [0x22],
+            version: .version3,
+            value: false
+        )
+        
+        XCTAssertEqual(
+            frameBytes,
+            [
+                0x22, // Identifier (See above.)
+                0x00, 0x00, 0x00, 0x28, // Size -- where do I get this?
+                0x00, 0x00, // Flags
+                0x01, // UCS‐2 -- what is this? version?
+                0x00, 0x00, 0x00, 0x00 // “false”
+            ]
+            // fails XCTAssertEqual failed: ("[34, 17, 0, 0, 0, 0, 0]") is not equal to ("[34, 0, 0, 0, 40, 0, 0, 1, 0, 0, 0, 0]") ugh
```

Comment:
Resolved the other comment and making a new one since I've made some changes. I just don't understand where some of these bytes are from or how to get them.

Reply:
- The identifier originates from line 20: `frameIdentifier: [0x22],`
- The size is being decided by the `MockFrameContentSizeCalculator()` from line 14. (See its implementation.)
- The flags are decided by the `MockFrameFlagsCreator()` from line 15. (See its implementation.)
- UCS‐2 is a string encoding. I don’t think you should be expecting it?
- The bytes for “false” are whatever you’ve made.

`XCTAssertEqual` is printing the bytes in decimal, but the `0x` is writing them in hexadecimal. so the `34` and the `0x22` are the same. macOS’s calculator application has a programmer mode (Menu Bar → View → Programmer) where you can enter a number and toggle between base 10 and 16 in the top right.

Reply:
Yup. “Mock” meaning “pretending to be a” size calculator. There is a time and place for that sort of thing (such as to remove network latency during testing), but upstream has gone way overboard in my opinion.

Reply:
ah ok. That makes more sense then. I've been trying for weeks to equate what the specification says to all these bytes I've been seeing in the code here and not managing it, probably in part because half the time there are bytes missing or that have nothing to do with what should actually be there.